build_tools/azure/python_nogil_requirements.txt

```diff
@@ -6,13 +6,12 @@
 # the latest cython will be picked up from PyPI, rather than the one from the
 # python-nogil index
 matplotlib
-# 2023-07-15 contourpy 1.1 needs meson which needs a recent pybind version
-# which is not available for python/nogil ...
-contourpy<1.1
 numpy
 scipy
 cython
 joblib
 threadpoolctl
-pytest
+# Avoid pytest 8.0.1 bug
+# https://github.com/pytest-dev/pytest/issues/12011
+pytest==8.0.0
```

Comment:
I merged https://github.com/scikit-learn/scikit-learn/pull/28475 so this can be reverted?

```suggestion
pytest
```

Reply:
Great thanks for this, I reverted this change now.